openmc/mgxs/groups.py

```diff
@@ -0,0 +1,275 @@
+from collections import Iterable
+from numbers import Real, Integral
+import copy
+import sys
+
+import numpy as np
+
+import openmc.checkvalue as cv
+
+
+if sys.version_info[0] >= 3:
+    basestring = str
+
+
+class EnergyGroups(object):
+    """An energy groups structure used for multi-group cross-sections.
+
+    Parameters
+    ----------
+    group_edges : ndarray
+        The energy group boundaries [MeV]
+    num_groups : Integral
+        The number of energy groups
+
+    Attributes
+    ----------
+    group_edges : ndarray
+        The energy group boundaries [MeV]
+    num_groups : Integral
+        The number of energy groups
+
+    """
+
+    def __init__(self, group_edges=None, num_groups=None):
+        self._group_edges = None
+        self._num_groups = None
+
+        if group_edges is not None:
+            self.group_edges = group_edges
+        if num_groups is not None:
+            self.num_groups = num_groups
+
+    def __deepcopy__(self, memo):
+        existing = memo.get(id(self))
+
+        # If this is the first time we have tried to copy object, create copy
+        if existing is None:
+            clone = type(self).__new__(type(self))
+            clone._group_edges = copy.deepcopy(self.group_edges, memo)
+            clone._num_groups = self.num_groups
+
+            memo[id(self)] = clone
+
+            return clone
+
+        # If this object has been copied before, return the first copy made
+        else:
+            return existing
+
+    def __eq__(self, other):
+        if not isinstance(other, EnergyGroups):
+            return False
+        elif self.group_edges != other.group_edges:
+            return False
+        else:
+            return True
+
+    def __ne__(self, other):
+        return not self == other
+
+    def __hash__(self):
+        return hash(tuple(self.group_edges))
+
+    @property
+    def group_edges(self):
+        return self._group_edges
+
+    @property
+    def num_groups(self):
+        return self._num_groups
+
+    @group_edges.setter
+    def group_edges(self, edges):
+        cv.check_type('group edges', edges, Iterable, Real)
+        cv.check_greater_than('number of group edges', len(edges), 1)
+        self._group_edges = np.array(edges)
+        self._num_groups = len(edges)-1
+
+    def generate_bin_edges(self, start, stop, num_groups, spacing='linear'):
```

Comment:
Again, this was a relic from an earlier time. I can't really see any good use case for this now, and agree that it wouldn't save any lines of code for the user anyway, so it is now removed.